src/proxy/reconnect.rs

```diff
@@ -1,15 +1,26 @@
-use std::fmt;
-
 use futures::{task, Async, Future, Poll};
+use std::fmt;
+use std::marker::PhantomData;
 use tower_reconnect;
 
 use svc;
 
+#[derive(Debug)]
+pub struct Layer<T, M> {
```

Comment:
Wasn't sure if it was necessary for all the types in this module to be generic over both `T` and `M`, but it seems like it is.

Reply:
Yeah, I tried and failed to remove it :man_shrugging: